frontend/src/store/FamLoginUserState.ts

```diff
@@ -207,7 +207,24 @@ const getCachedAppRolesForDelegatedAdmin = (
             ? -1
             : 1;
     });
-}
+};
+
+const getMyCachedPermissions = () => {
+    let myRoles: any = [];
+    getUserAccess()?.map((item) => {
+        item.grants.map((grant) => {
+            grant.roles?.forEach((role) => {
+                Object.assign(grant.application, {
+                    roles: role,
+                });
+                myRoles.push({
+                    application: grant.application,
+                });
+            });
+        });
+    });
+    return myRoles;
```

Comment:
For example your return data can be in the format of 
```
{
application: {id: 1, name: 'FAM', description: 'Forests Access Management', env: null},     // this is exact same as the application info in user's access
role: {id: 4, name: 'Reviewer',type_code: 'C','forest_clients': null} // this is also the same as the role info in user's access
}
```
For FAM_ADMIN and APP_ADMIN case, you just add a role field, `role: {name: 'Admin'}`

Reply:
got it! Thanks!